src/ripple/app/main/GRPCServer.h

```diff
@@ -0,0 +1,241 @@
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CORE_GRPCSERVER_H_INCLUDED
+#define RIPPLE_CORE_GRPCSERVER_H_INCLUDED
+
+#include <ripple/app/main/Application.h>
+#include <ripple/core/JobQueue.h>
+#include <ripple/net/InfoSub.h>
+#include <ripple/protocol/ErrorCodes.h>
+#include <ripple/resource/Charge.h>
+#include <ripple/rpc/Context.h>
+#include <ripple/rpc/GRPCHandlers.h>
+#include <ripple/rpc/Role.h>
+#include <ripple/rpc/impl/Handler.h>
+#include <ripple/rpc/impl/RPCHelpers.h>
+#include <ripple/rpc/impl/Tuning.h>
+
+#include "rpc/v1/xrp_ledger.grpc.pb.h"
+#include <grpcpp/grpcpp.h>
+
+namespace ripple {
+
+// Interface that CallData implements
+class Processor
+{
+public:
+    virtual ~Processor() = default;
+
+    // process a request that has arrived. Can only be called once per instance
+    virtual void
+    process() = 0;
+
+    // abort processing this request. called when server shutsdown
+    virtual void
+    abort() = 0;
+
+    // create a new instance of this CallData object, with the same type
+    //(same template parameters) as original. This is called when a CallData
+    // object starts processing a request. Creating a new instance allows the
+    // server to handle additional requests while the first is being processed
+    virtual std::shared_ptr<Processor>
+    clone() = 0;
+
+    // true if this object has finished processing the request. Object will be
+    // deleted once this function returns true
+    virtual bool
+    isFinished() = 0;
+};
+
+class GRPCServerImpl final
+{
+private:
+    // CompletionQueue returns events that have occurred, or events that have
+    // been cancelled
+    std::unique_ptr<grpc::ServerCompletionQueue> cq_;
+
+    std::vector<std::shared_ptr<Processor>> requests_;
+
+    // The gRPC service defined by the .proto files
+    rpc::v1::XRPLedgerAPIService::AsyncService service_;
+
+    std::unique_ptr<grpc::Server> server_;
+
+    Application& app_;
+
+    std::string serverAddress_;
+
+    // typedef for function to bind a listener
+    // This is always of the form:
+    // rpc::v1::XRPLedgerAPIService::AsyncService::Request[RPC NAME]
+    template <class Request, class Response>
+    using BindListener = std::function<void(
+        rpc::v1::XRPLedgerAPIService::AsyncService&,
+        grpc::ServerContext*,
+        Request*,
+        grpc::ServerAsyncResponseWriter<Response>*,
+        grpc::CompletionQueue*,
+        grpc::ServerCompletionQueue*,
+        void*)>;
+
+    // typedef for actual handler (that populates a response)
+    // handlers are defined in rpc/GRPCHandlers.h
+    template <class Request, class Response>
+    using Handler = std::function<std::pair<Response, grpc::Status>(
+        RPC::GRPCContext<Request>&)>;
+
+public:
```

Comment:
Recommend adding:
```
    GRPCServerImpl(const GRPCServerImpl&) = delete;
    GRPCServerImpl& operator=(const GRPCServerImpl&) = delete;

```
This both changes and clarifies the special move members. Without this change GRPCServerImpl is MoveConstructible, which is never used.